lib/access_import/haupt_person.rb

```diff
@@ -0,0 +1,63 @@
+class HauptPerson
+  include AccUtils
+  attr_reader :record, :records
+
+  def initialize(acimport, accid = nil)
+    @laender = acimport.laender
+    @acdb = acimport.acdb
+    @plz = acimport.plz
+    @sprache_self = acimport.sprache_hauptperson
+    if accid
+      @record = internal_find(accid)
+      return self
+    end
+    @records = all
+    self
+  end
+
+  def internal_find(id)
+    if @records
+      @records[id.to_s]
+    else
+      person = @acdb[:tbl_Hauptpersonen].select do |hp|
+        hp[:pk_Hauptperson] == id.to_s
+      end.first
```

Comment:
select returns an array, even if there is only one match. So with first I free it from beeing nested as the only element inside an array.